### PR TITLE
Tabs don't display items other than 0 and 1

### LINKED ISSUE #1119: Tabs don't display items other than 0 and 1
I'm guessing that this change caused the problem
https://github.com/flutter/flutter/pull/1098

You can see the problem in the material gallery tabs demos.

### PR DESCRIPTION
**Pull request by abarth**
The TabBarView wants to remap page indices during tab transitions, which
I broke when I refactored TabBarView.

Fixes #1119

### CODE REVIEW

**Comment by abarth** [comment]
@HansMuller

File: packages/flutter/lib/src/material/tabs.dart [hunk 1]
```diff
@@ -904,14 +910,16 @@ class _TabBarViewState<T> extends PageableListState<TabBarView<T>> implements Ta
 
     if (scrollVelocity.dx.abs() > _kMinFlingVelocity) {
       final int selectionDelta = scrollVelocity.dx > 0 ? -1 : 1;
-      _selection.value = _selection.values[(_selection.index + selectionDelta).clamp(0, _tabCount - 1)];
+      int targetIndex = (_selection.index + selectionDelta).clamp(0, _tabCount - 1);
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/tabs.dart]: This should probably be final.  Likewise for the next one.

File: packages/flutter/lib/src/material/tabs.dart [hunk 2]
```diff
@@ -833,17 +832,26 @@ class _TabBarViewState<T> extends PageableListState<TabBarView<T>> implements Ta
     _selection = null;
   }
 
+  void _updateItems(int first, int second, [int third]) {
+    List<Widget> widgets = config.children;
+    _items = new List<Widget>();
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/tabs.dart]: Couldn't this just be `_items = <Widget>[widgets[first], widgets[second]]` ?

Comment (abarth) [inline packages/flutter/lib/src/material/tabs.dart]: Yeah.

**Comment by HansMuller** [comment]
LGTM. Sorry this stuff isn't simpler.

**Pull request merged**